File: micronaut/src/main/java/blog/bocharoviliyav/service/PersonService.java
package blog.bocharoviliyav.service;


import blog.bocharoviliyav.model.Person;
import blog.bocharoviliyav.repo.PersonRepo;
import jakarta.inject.Singleton;

@Singleton
public class PersonService {

  private final PersonRepo personRepo;

  PersonService(PersonRepo personRepo) {
    this.personRepo = personRepo;
  }

  public Person createOrUpdate(Person person) {
    return personRepo.createOrUpdate(person);
  }

  public Person findById(Integer id) {
    return personRepo.findById(id);
  }


}

File: golang/personApp/service/PersonService.go
package service

import (
	"golang/personApp/model"
	"golang/personApp/repo"
)

type PersonService interface {
	GetPersonByID(id int) (*model.Person, error)
	SavePerson(person *model.Person) error
}

type personService struct {
	repo repo.PersonRepository
}

func NewPersonService(repo repo.PersonRepository) PersonService {
	return &personService{repo: repo}
}

func (s *personService) GetPersonByID(id int) (*model.Person, error) {
	return s.repo.GetPersonByID(id)
}

func (s *personService) SavePerson(person *model.Person) error {
	return s.repo.SavePerson(person)
}
